community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/Eagerness.scala

```diff
@@ -107,7 +107,8 @@ object Eagerness {
   def headWriteReadEagerize(inputPlan: LogicalPlan, query: PlannerQuery)
                        (implicit context: LogicalPlanningContext): LogicalPlan = {
     val alwaysEager = context.config.updateStrategy.alwaysEager
-    if (alwaysEager || query.tail.isDefined && writeReadConflictInHead(query, query.tail.get))
+    val conflictInHorizon = query.queryGraph.horizonOverlap(query.horizon)
+    if (alwaysEager || conflictInHorizon || query.tail.isDefined && writeReadConflictInHead(query, query.tail.get))
```

Comment:
Do you mean when the horizon is in a tail and conflicts with writes made in a `PlannerQuery` one or several hops earlier?